geonode/upload/uploadhandler.py

```diff
@@ -0,0 +1,215 @@
+        """
+        We use the content_length to signal whether or not this handler should be used.
+        """
+        # Check the content-length header and url_name to see if we should
+        self.is_view_elegible_for_size_restriction = (
+            self.request.resolver_match and self.request.resolver_match.url_name in self.elegible_url_names
+        )
+        self.max_size_allowed = 0
+        self.activated = False
+
+        # If the post is too large, we create a empty UploadedFile, otherwise another handler will take care or it.
+        if self.is_view_elegible_for_size_restriction:
+            self.max_size_allowed = self._get_max_size()
+            self.activated = content_length > self.max_size_allowed
+            if self.activated:
+                return self._handle_raw_input_without_file_stream(input_data, META, content_length, boundary, encoding)
+
+    def _handle_raw_input_without_file_stream(self, input_data, META, raw_content_length, boundary, encoding=None):
+        """
+        Replaces django.http.multipartparser.MultiPartParser.parse
+        A rfc2388 multipart/form-data parser but replacing the file stream to the creation of empty files.
+        Returns a tuple of ``(MultiValueDict(POST), MultiValueDict(FILES))``.
+        """
+        # Create the data structures to be used later.
+        _post = QueryDict(mutable=True)
+        _files = MultiValueDict()
+
+        # For compatibility with low-level network APIs (with 32-bit integers),
+        # the chunk size should be < 2^31, but still divisible by 4.
+        _chunk_size = min([2 ** 31 - 4, self.chunk_size])
+
+        # Instantiate the parser and stream:
+        stream = LazyStream(ChunkIter(input_data, _chunk_size))
+
+        # Whether or not to signal a file-completion at the beginning of the loop.
+        old_field_name = None
+
+        # Number of bytes that have been read.
+        num_bytes_read = 0
+        # To count the number of keys in the request.
+        num_post_keys = 0
+        # To limit the amount of data read from the request.
+        read_size = None
+
+        for item_type, meta_data, field_stream in Parser(stream, boundary):
+            if old_field_name:
+                # We run this at the beginning of the next loop
+                # since we cannot be sure a file is complete until
+                # we hit the next boundary/part of the multipart content.
+                file_obj = self.file_complete(raw_content_length)
+                if file_obj:
+                    # If it returns a file object, then set the files dict.
+                    _files.appendlist(force_str(old_field_name, encoding, errors="replace"), file_obj)
+                old_field_name = None
+
+            try:
+                disposition = meta_data["content-disposition"][1]
+                field_name = disposition["name"].strip()
+            except (KeyError, IndexError, AttributeError):
+                continue
```

Comment:
can we log the exception?

Reply:
Usually this is not logged by the default django managers. We can log, but should we log in this case?

https://github.com/django/django/blob/main/django/http/multipartparser.py#L166-L169